l10n_fr_hr_rup/models/hr_contract.py

```diff
@@ -7,14 +7,31 @@ class HrContract(models.Model):
     _inherit = "hr.contract"
     _order = "date_start,name"
 
-    pcs_id = fields.Many2one("hr.employee.pcs", "PCS")
     employer_address_id = fields.Many2one("res.partner", "Addresse de l'employeur")
-    qualification = fields.Char("Qualification")
-    work_location = fields.Char("Localisation du bureau")
+    pcs_id = fields.Many2one(
+        "hr.employee.pcs", "PCS", compute="_compute_pcs_id", store=True, readonly=False
+    )
+    qualification = fields.Char(
+        "Qualification", compute="_compute_qualification", store=True, readonly=False
+    )
+    work_location = fields.Char(
+        "Localisation du bureau",
+        compute="_compute_work_location",
+        store=True,
+        readonly=False,
+    )
 
-    @api.onchange("employee_id")
-    def _onchange_employee_id(self):
-        super()._onchange_employee_id()
-        self.pcs_id = self.employee_id.pcs_id
-        self.qualification = self.employee_id.qualification
-        self.work_location = self.employee_id.work_location
+    @api.depends("employee_id")
+    def _compute_pcs_id(self):
+        for rec in self:
+            rec.pcs_id = rec.employee_id.pcs_id
+
+    @api.depends("employee_id")
+    def _compute_qualification(self):
+        for rec in self:
+            rec.qualification = rec.employee_id.qualification
+
+    @api.depends("employee_id")
+    def _compute_work_location(self):
```

Comment:
I think this kind of information is susceptible to change; by computing it we lose historic data on the contracts

Reply:
Sorry, that's not actually the case since it is computed with store=True. This makes it more user-friendly but also easier to make mistakes IMHO

Reply:
These are computed stored writeable fields. From the user perspective / functional side, there's no difference with the onchange.

Can you elaborate on your concerns?

Reply:
Yes, that's my mistake. I thought about some client making a mistake by changing employee_id but thinking about it that's out of the scope of this PR, I switch to approve now to avoid blocking